### ISSUE TITLE
Can't set expected result size with MultiSearch

### ISSUE THREAD

**Issue by Ehud-Azikri** [searchbox-io/Jest#212]
Hi,

I'm trying to use the MultiSearch class to generate a few query requests together - in my case a few more like this requests.
I want to to set the size of the response for each request to some value e.g. 100, but every response comes back with (at most) only 10 hits, which is the default value.

my code looks like this:

       List<Search> searchList = new ArrayList<>();
       for (String docId : docIds) {

            String query = String.format("{\"more_like_this\" : {" +"\"fields\" : [\"%s\"]," +
                    "        \"docs\" : [" +
                    "        {" +
                    "            \"_index\" : \"%s\"," +
                    "            \"_type\" : \"%s\"," +
                    "            \"_id\" : \"%s\"" +
                    "        }]," +
                    "       \"min_term_freq\" : 1," +
                    "       \"min_doc_freq\" : 1," +
                    "        \"max_query_terms\": %s"+
                    "    }}",fields,index,type,docId,Integer.toString(numTerms));

            Search search = new Search.Builder(query)
                    .addIndex(index)
                    .addType(type)
                    .setParameter("size", size)
                    .build();

            searchList.add(search);
        }
        MultiSearch multiSearch = new MultiSearch.Builder(searchList).build();
        JestResult result = jestClient.execute(multiSearch);

Appreciating any help
Ehud

**Comment by kramer** [comment]
[In Multi Search `size` should be set on each query payload](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html) (**not** as HTTP parameter).

So your code should look like: 

```java
List<Search> searchList = new ArrayList<>();
   for (String docId : docIds) {

        String query = String.format("{"query" :{\"more_like_this\" : {" +"\"fields\" : [\"%s\"]," +
                "        \"docs\" : [" +
                "        {" +
                "            \"_index\" : \"%s\"," +
                "            \"_type\" : \"%s\"," +
                "            \"_id\" : \"%s\"" +
                "        }]," +
                "       \"min_term_freq\" : 1," +
                "       \"min_doc_freq\" : 1," +
                "        \"max_query_terms\": %s"+
                "    }}, \"size\": %d}",fields,index,type,docId,Integer.toString(numTerms), size);

        Search search = new Search.Builder(query)
                .addIndex(index)
                .addType(type)
                .build();

        searchList.add(search);
    }
```

**Comment by Ehud-Azikri** [comment]
Thanks Cihat for the quick reply!

I implemented it and it usually works ok. However I did some benchmark test that takes 1000 document IDs and looks for the result of theMLT search queries – multiple and single.
I then compared the results.
For 5 document IDs, the number of hits using single and multiple queries was different. In four cases the difference was minor (a difference of 1-2 hits between multiple and single queries). In one case it was significant – 100 hits were returned in multiple query but only 8 In single query.
In two other cases the same number of hits returned – 100 (this was the “size” parameter) - but results were different – there was an overlap of 85 hits, 15 were different.

I ran the test again and results repeat.

May I ask if you have some speculations why these differences exists?

Thanks again
Ehud
